esx_service/vmdk_ops.py

```diff
@@ -1670,22 +1683,39 @@ def set_vol_opts(name, tenant_name, options):
 
     return False
 
+def wait_ops_in_flight():
+    global STOP_RETRY_COUNT
+
+    # Set the STOP flag to true
+    stopEvent.set()
+
+    # Wait until the in-flight ops counter goes down to zero
+    while ((opsCounter.value > 0) and (STOP_RETRY_COUNT > 0)):
+        logging.info("Service is being stopped. Waiting for completion of %d operations in flight...", opsCounter.value)
+        time.sleep(STOP_RETRY_SLEEP)
+        STOP_RETRY_COUNT -= 1
+
+    if (opsCounter.value == 0):
```

Comment:
how you can guarantee that one it goes to 0 and returns 0 here, nobody starts a new op between now and the line 1699?

Reply:
As @pshahzeb mentioned, all new incoming requests should be rejected by line 1857. Basically, once the flag is set, the counter should just go down, and never go up.

Reply:
Correct, it is not possible in the current implementation - but it depends on other parts which are outside of the function and are not clear here. At the very least a comment is required.  
At any rate , this is not very important as this code needs to change (see my other comment about blocking main thread)